src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +225,296 @@ def execute(self, reqStatus):
         # start threads in MSManager which should call this method
         summary = dict(RULECLEANER_REPORT)
 
+        self.currThread = current_thread()
+        self.currThreadIdent = self.currThread.name
+        self.updateReportDict(summary, "thread_id", self.currThreadIdent)
+
+        msg = "{}: MSRuleCleaner is running in mode: {}.".format(self.currThreadIdent, self.mode)
+        self.logger.info(msg)
+
+        # Build the list of workflows to work on:
         try:
-            requestRecords = self.getRequestRecords(reqStatus)
-            self.updateReportDict(summary, "total_num_requests", len(requestRecords))
-            msg = "  retrieved %s requests. " % len(requestRecords)
-            msg += "Service set to process up to %s requests per cycle." % self.msConfig["limitRequestsPerCycle"]
+            requestRecords = {}
+            for status in reqStatus:
+                requestRecords.update(self.getRequestRecords(status))
             self.logger.info(msg)
         except Exception as err:  # general error
             msg = "Unknown exception while fetching requests from ReqMgr2. Error: %s", str(err)
             self.logger.exception(msg)
             self.updateReportDict(summary, "error", msg)
 
-        # this one is put here just for example.
-        self.updateReportDict(summary, "error", 42)
+        # Call _execute() and feed the relevant pipeline with the objects popped from requestRecords
+        try:
+            clean_num_requests, total_num_requests, archived_num_requests = self._execute(requestRecords)
```

Comment:
I used that  name intentionally, because this is actually an auxiliary function to `execute()`. It is basically used to do the job for the top level execute function, but only this time it splits the work by workflow instead of working on bulk.